src/components/VDatePicker/VDatePicker.js

```diff
@@ -178,29 +184,50 @@ export default {
     pickerDate (val) {
       if (val) {
         this.tableDate = val
-      } else if (this.value && this.type === 'date') {
-        this.tableDate = this.sanitizeDateString(this.value, 'month')
-      } else if (this.value && this.type === 'month') {
-        this.tableDate = this.sanitizeDateString(this.value, 'year')
+      } else if (this.lastValue && this.type === 'date') {
+        this.tableDate = this.sanitizeDateString(this.lastValue, 'month')
+      } else if (this.lastValue && this.type === 'month') {
+        this.tableDate = this.sanitizeDateString(this.lastValue, 'year')
       }
     },
     value () {
+      this.checkMultipleProp()
       this.setInputDate()
+
       if (this.value && !this.pickerDate) {
         this.tableDate = this.sanitizeDateString(this.inputDate, this.type === 'month' ? 'year' : 'month')
       }
     },
     type (type) {
       this.activePicker = type.toUpperCase()
 
-      if (this.value) {
-        const date = this.sanitizeDateString(this.value, type)
-        this.$emit('input', this.isDateAllowed(date) ? date : null)
+      if (this.value && this.value.length) {
+        const output = (this.multiple ? this.value : [this.value])
+          .map(val => this.sanitizeDateString(val, type))
+          .filter(this.isDateAllowed)
+        this.$emit('input', this.multiple ? output : output[0])
       }
     }
   },
 
   methods: {
+    emitInput (newInput) {
+      const output = this.multiple
+        ? (
+          this.value.indexOf(newInput) === -1
+            ? this.value.concat([newInput])
+            : this.value.filter(x => x !== newInput)
+        )
+        : newInput
+
+      this.$emit('input', output)
+      this.multiple || this.$emit('change', newInput)
+    },
+    checkMultipleProp () {
+      if (this.multiple ^ Array.isArray(this.value)) {
+        consoleWarn(`Value must be an Array/String (depending on multiple), got ${typeof this.value}`, this)
```

Comment:
- Warn based on what is actually expected with the provided props
- `value.constructor.name` gives a better result, might need a null check though